pants

```diff
@@ -116,7 +116,15 @@ function determine_default_python_exe {
     supported_message='3.6+'
   fi
   for version in "${supported_python_versions[@]}"; do
-    command -v "python${version}" && return 0
+    interpreter_path="$(command -v "python${version}")"
+    if [[ -z "${interpreter_path}" ]]; then
+      continue
+    fi
+    # Check if the Python version is installed via Pyenv but not activated.
```

Comment:
This was a great step forward - we no longer error.

Better would be to be able ignore shims, but that's hard to do. Best would be to be fully pyenv aware as pants is and skip the shims (via your fix here) and use pyenv pythons even when not activated via a `$PYENV_ROOT/versions/*/bin/python${version}` scan: https://github.com/pantsbuild/pants/blob/c8243f1ca66d00495033783eacfe4c02d76dcfae/src/python/pants/python/python_setup.py#L265-L290

Reply:
> Better would be to be able ignore shims, but that's hard to do. Best would be to be fully pyenv aware and use pyenv pythons even when not activated

The idea being that it's an implementation detail which Python interpreter is used to run Pants, so it shouldn't matter if a user has disabled the interpreter with `pyenv`. That does not necessarily signal intent that they don't want Pants using that particular interpreter?

There is also always a safety valve via `$PYTHON`.